src/DI/ElasticaExtension.php

```diff
@@ -20,29 +20,37 @@ class ElasticaExtension extends CompilerExtension
 
 	public function getConfigSchema(): Schema
 	{
-		// https://elastica-docs.readthedocs.io/en/latest/client.html#client-configurations
+		// https://github.com/ruflin/Elastica/blob/master/src/ClientConfiguration.php#L26
+		$clientConfig = [
+			'host' => Expect::string()->nullable()->dynamic(),
+			'port' => Expect::int()->nullable()->dynamic(),
+			'path' => Expect::string()->nullable(),
+			'url' => Expect::string()->nullable(),
+			'proxy' => Expect::string()->nullable(),
+			'transport' => Expect::string()->nullable(),
+			'compression' => Expect::bool(),
+			'persistent' => Expect::bool(),
+			'timeout' => Expect::int()->nullable(),
+			'retryOnConflict' => Expect::int(),
+			'bigintConversion' => Expect::bool(),
+			'username' => Expect::string()->nullable()->dynamic(),
+			'password' => Expect::string()->nullable()->dynamic(),
+			'auth_type' => Expect::anyOf('basic', 'digest', 'gssnegotiate', 'ntlm')->nullable()->dynamic(),
+			'curl' => Expect::arrayOf('mixed', 'int'),
+			'headers' => Expect::arrayOf('string', 'string'),
+		];
+
 		return Expect::structure([
 			'debug' => Expect::bool(false),
-			'hosts' => Expect::arrayOf(
-				Expect::structure([
-					'port' => Expect::int(),
-					'path' => Expect::string(),
-					'host' => Expect::string(),
-					'url' => Expect::string(),
-					'proxy' => Expect::string(),
-					'transport' => Expect::string(),
-					'persistent' => Expect::bool(),
-					'timeout' => Expect::int(),
-					'connections' => Expect::array(),
+			'config' => Expect::structure(array_merge(
+				$clientConfig,
+				[
+					'connections' => Expect::arrayOf(
```

Comment:
Why is it merged here? Can we skip merging?

Reply:
Because a client config contains 16 options and the options are present in the root `config` and also under the key `config.connections[]`. 
Merging came to be better for me. But yeah if you insist we can skip merging and define the options duplicated.